lib/vimgolf/challenge.rb

```diff
@@ -34,9 +55,14 @@ def download
         @data['in']['data'].gsub!(/\r\n/, "\n")
         @data['out']['data'].gsub!(/\r\n/, "\n")
 
-        @type = @data['in']['type']
+        # be sure to sanitize the types
+        @type = @data['in']['type'].gsub(/[^\w-]/, '.')
```

Comment:
`shellescape` is not portable. The main purpose of this `gsub` is actually to eliminate directory separators, which `shellescape` doesn't do. In fact, it adds a bunch of backslashes, which is a directory separator in Windows.